_posts/2015-03-16-CocoaPods.org-Take-Two.md

```diff
@@ -0,0 +1,151 @@
+
+Once [CocoaDocs v2][3] launched in May 2014, I set my sights on trying to move cocoapods.org a little bit closer to my larger vision of what the website would provide.
+
+In an ideal world, you should be able to find a library that fits your constraints using just the front page of CocoaPods.org. Often there is more than one library, and that’s where things gets a bit more complex. In previous versions of the site, in order to choose between multiple libraries, you need to open them up on GitHub in separate tabs and compare. Now, you don’t need to. 🎉
+
+## Here’s how it works
+
+Let’s take a random pod. [MiniFuture][4]. This is what happens after a Pod is released:
+
+* A new Podspec version is added to [Trunk][5]
+* A webhook from trunk triggers a reindex of the pod in the search engine which uses the generated metadata to sort and display pod results.
+* A another webhook gets sent to CocoaDocs triggering a documentation build
+* The documentation build uses [Jazzy][jazzy] ( or [appledoc][appledoc] for Objective-C projects ) to generate documentation. 
+* After documentation has finished CocoaDocs starts to generating metadata. This provides us with a rich dataset to start providing some useful statistics. This is centralized in [metrics.cocoapods.org][6], where we currently hold up-to-date GitHub metrics also.
+
+```json
+// http://metrics.cocoapods.org/api/v1/pods/MiniFuture
+
+{
+	"cocoadocs": {
+	    "created_at": "2015-03-08 16:48:25 UTC",
+	    "doc_percent": 22,
+	    "dominant_language": "Swift",
+	    "download_size": 416,
+	    "initial_commit_date": "2015-03-01 12:03:09 UTC",
+	    "license_canonical_url": "http://opensource.org/licenses/MIT",
+	    "license_short_name": "MIT",
+	    "notability": 0,
+	    "quality_estimate": 50,
+	    "readme_complexity": 72,
+	    "rendered_readme_url": "http://cocoadocs.org/docsets/MiniFuture/0.1.0/README.html",
+	    "total_comments": 8,
+	    "total_files": 4,
+	    "total_lines_of_code": 327,
+	    "total_test_expectations": 204,
+	    "updated_at": "2015-03-08 16:48:25 UTC"
+	},
+	"github": {
+	    "contributors": 1,
+	    "created_at": "2015-03-01 10:09:08 UTC",
+	    "forks": 0,
+	    "language": "Swift",
+	    "open_issues": 0,
+	    "open_pull_requests": 0,
+	    "stargazers": 8,
+	    "subscribers": 2,
+	    "updated_at": "2015-03-05 01:01:44 UTC"
+	}
+}
+```
+* A preview image is generated for the library for Social Media, chat, or really anything that supports open graph previews.
+  {% breaking_image http://cocoadocs.org/docsets/MiniFuture/0.1.0/preview.png %}
+  
+* We then generate an estimated quality number for the library. This is based on a collection of [individual metrics][7] that are applied to the generated metrics data. They take a library and add or remove points based whether the rule applies for that library. These range from the popularity of the library, total test expectations / lines of code to the files, and average lines of code per file in the library.
```

Comment:
It's the Oxford comma!

-Samuel E. Giddins

On Mar 17, 2015, at 8:26 AM, Laura Brown <notifications@github.com> wrote:

In _posts/2015-03-16-CocoaPods.org-Take-Two.md:

> +	    "contributors": 1,
> +	    "created_at": "2015-03-01 10:09:08 UTC",
> +	    "forks": 0,
> +	    "language": "Swift",
> +	    "open_issues": 0,
> +	    "open_pull_requests": 0,
> +	    "stargazers": 8,
> +	    "subscribers": 2,
> +	    "updated_at": "2015-03-05 01:01:44 UTC"
> +	}
> +}
> +```
> +* A preview image is generated for the library for Social Media, chat, or really anything that supports open graph previews.
> +  {% breaking_image http://cocoadocs.org/docsets/MiniFuture/0.1.0/preview.png %}
> +  
> +* We then generate an estimated quality number for the library. This is based on a collection of [individual metrics][7] that are applied to the generated metrics data. They take a library and add or remove points based whether the rule applies for that library. These range from the popularity of the library, total test expectations / lines of code to the files, and average lines of code per file in the library.
"files, and" comma not necessary

—
Reply to this email directly or view it on GitHub.